lib/azure/armrest/model/base_model.rb

```diff
@@ -33,10 +33,21 @@ class BaseModel < Delegator
       #
       def initialize(json)
         @json = json
+        @resource_group = nil
         @ostruct = JSON.parse(json, object_class: OpenStruct)
         __setobj__(@ostruct)
       end
 
+      # Return the resource group for the current object.
+      def resource_group
+        @resource_group ||= id[/resourceGroups\/(.+?)\//i, 1] rescue nil
+      end
+
+      # Return a hash of tags associated with the resource.
+      def tags
```

Comment:
After further review, I think this is more difficult to implement than it's worth. For now I think we should just define methods explictly as we need them.